File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int depth = 0;
    public int maxdepth = 0;
    public int result = 0;
    public void deepestSum(TreeNode node) {
        if (node.left == null && node.right == null) {
            if(depth > maxdepth) {
                maxdepth = depth;
                result = node.val;
            }
            else if(depth == maxdepth) {
                result += node.val;
            }
            return;
        }
        if(node.left != null) {
            depth++;
            deepestSum(node.left);
            depth--;
        }
        if(node.right != null) {
            depth++;
            deepestSum(node.right);
            depth--;
        }
    }
    public int deepestLeavesSum(TreeNode root) {
        deepestSum(root);
        return result;
    }
}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import defaultdict
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        depth, maxdepth = 0, 0
        ans = 0
        queue = [(root, depth)]
        
        while queue:
            node, depth = queue.pop(0)
            
            if node.left == None and node.right == None:
                if depth > maxdepth:
                    maxdepth = depth
                    ans = node.val
                elif depth == maxdepth:
                    ans += node.val
            if node.left:
                queue.append((node.left, depth+1))
            if node.right:
                queue.append((node.right, depth+1))
        return ans
                